### PR TITLE
Add components for converting satwnd amv goes data from bufr dump to ioda.  

### PR DESCRIPTION
**Pull request by emilyhcliu**
Add components for converting satwnd amv goes data from bufr dump to ioda.

dump/mapping
- bufr2ioda mapping file
- bufr2ioda Python converter

dump/config
- IODA configuration test YAML for script backend
- IODA configuration test YAML for bufr backend

/ush/test 
- bufr2ioda.sh
- README.md

Notes:  The test shell script (bufr2ioda.sh) can test the following for each obs type:
- bufr2netcdf ---> convert bufr to NetCDF IODA format using bufr-query with mapping file
- script2netcdf ---> convert bufr to NetCDF IODA format using bufr-query with mapping file and python interface
- bufr_backend ---> convert bufr to IODA in IODA backend with mapping file
- script_backend ---> convert bufr to IODA in IODA backend with mapping file and Python script

### CODE REVIEW

File: ush/test/README.md
Comment (rmclaren) [inline ush/test/README.md]: bufr2ioda.sh

File: ush/test/README.md
Comment (rmclaren) [inline ush/test/README.md]: bufr2ioda.sh

**Review by rmclaren — DISMISSED** [review]

File: dump/config/bufr2ioda_bufr_backend_satwnd_amv_goes.yaml
Comment (CoryMartin-NOAA) [inline dump/config/bufr2ioda_bufr_backend_satwnd_amv_goes.yaml]: just noting that we will want to template these if they are not in a fixed 'test' directory

File: ush/test/README.md
Comment (CoryMartin-NOAA) [inline ush/test/README.md]: Let me include wxflow in the obsForge clone

File: ush/test/bufr2ioda.sh
Comment (CoryMartin-NOAA) [inline ush/test/bufr2ioda.sh]: general question, @emilyhcliu , do we want to set up `ctests` so that a user just needs to run `ctest` to test like we do for JEDI repos?

Comment (emilyhcliu) [reply ush/test/bufr2ioda.sh]: Yes.  We will add CTests in obsForge.

File: ush/test/README.md
Comment (xincjin-NOAA) [inline ush/test/README.md]: Can we use named inputs? Otherwise, in order to give the final input we have to type all others.

**Review by rmclaren — DISMISSED** [review]
@rmclaren Should the script "bufr2ioda.sh" be called "bufr2netcdf.sh"?
This test script can handle four cases with or without MPI.  
1. bufr2netcdf (bufr-query)
2. scrtip2netcdf (bufr-query)
3. bufr_backend (bufr-query+ioda)
4. script2_backend (bufr-query+ioda)

It covers tests for bufr-query and ioda.  Do you have any suggestion for a better name?

This is a shell script to help team members test their converters.
We will add CTESTs for these tests in obsForge.

**Comment by rmclaren** [comment]
@emilyhcliu Dunno, maybe **encodeBufr.sh** or **runBufr.sh** (since we it for testing the second option might make sense (?).

**Comment by emilyhcliu** [comment]
> @emilyhcliu Dunno, maybe **encodeBufr.sh** or **runBufr.sh** (since we it for testing the second option might make sense (?).

I will use `encodeBufr.sh`.  Thanks.

File: dump/mapping/bufr2ioda_satwnd_amv_goes.py
Comment (nicholasesposito) [inline dump/mapping/bufr2ioda_satwnd_amv_goes.py]: Should we consider adding runtimes to the all subroutines in case we're curious if there are hang ups?

File: ush/test/README.md
Comment (nicholasesposito) [inline ush/test/README.md]: This should be encodeBufr.sh

File: ush/test/README.md
Comment (nicholasesposito) [inline ush/test/README.md:28]: Recommend adding here a line saying to mkdir a testinput directory in ```sorc/spoc/ush/test``` and also moving your input file here. The encodeBufr.sh does not do this (though it does make testoutput). Might be worth having encodeBufr.sh doing it too.

File: ush/test/encodeBufr.sh
Comment (nicholasesposito) [inline ush/test/encodeBufr.sh]: Suggest adding lines to move the appropriate files to the work_dir from their original locations. Written as is, the script looks for ioda_config_yaml, mapping_file, and input_file which are not already there.

Comment (emilyhcliu) [reply ush/test/encodeBufr.sh]: @CoryMartin-NOAA and @rmclaren Can we merge this branch?  
I have tested the files added thoroughly.

**Review by rmclaren — DISMISSED** [review]

File: ush/test/README.md
Comment (emilyhcliu) [reply ush/test/README.md]: Ha!  Thanks for reminding me.  I will change it now.

File: dump/config/bufr2ioda_bufr_backend_satwnd_amv_goes.yaml
Comment (emilyhcliu) [reply dump/config/bufr2ioda_bufr_backend_satwnd_amv_goes.yaml]: I missed this comment. 
This configuration file is for CTEST only.  So, I think a pre-defined time window should be OK.

File: dump/mapping/bufr2ioda_satwnd_amv_goes.py
Comment (emilyhcliu) [reply dump/mapping/bufr2ioda_satwnd_amv_goes.py]: Let's merge this as is.  We can make improvement later

**Review by CoryMartin-NOAA — DISMISSED** [review]
@emilyhcliu merge away if you're ready and happy

**Review by xincjin-NOAA — APPROVED** [review]
look good to me.

**Review by rmclaren — APPROVED** [review]

File: dump/mapping/bufr2ioda_satwnd_amv_goes.py
Comment (rmclaren) [inline dump/mapping/bufr2ioda_satwnd_amv_goes.py]: Since these are all required arguments you shouldn't need  to specify the -i --input, etc...

**Review by rmclaren — DISMISSED** [review]

File: dump/mapping/bufr2ioda_satwnd_amv_goes.py
Comment (emilyhcliu) [reply dump/mapping/bufr2ioda_satwnd_amv_goes.py]: Fixed!!

Comment (emilyhcliu) [reply dump/mapping/bufr2ioda_satwnd_amv_goes.py]: @rmclaren I made them to `positional argument` and remove the flags (-i, --input, ...etc)
Please approve again.  So, we can merge it.  Thanks.

**Comment by emilyhcliu** [comment]
@CoryMartin-NOAA Can you give both @rmclaren and @emilyhcliu authority to merge PR to SOPC?  
Don't merge this one yet.  We still have one very minor changes to do.

**Comment by CoryMartin-NOAA** [comment]
@emilyhcliu you should have already the ability to do so, let me 'approve again' do you see the merge button if I approve?

**Review by CoryMartin-NOAA — DISMISSED** [review]

**Review by CoryMartin-NOAA — DISMISSED** [review]

**Comment by emilyhcliu** [comment]
@CoryMartin-NOAA I committed all the changes.  Please approve again. Thanks...
At this point, `Merging is blocked`

**Review by CoryMartin-NOAA — APPROVED** [review]

**Comment by emilyhcliu** [comment]
@CoryMartin-NOAA Thanks.  I can merge the PR and I just did.  Thank you!

**Pull request merged**
